File: rust/abc198/src/c.rs
use proconio::input;
use std::io::{stdout, BufWriter, Write};

pub fn main() {
  let out = stdout();
  let mut out = BufWriter::new(out.lock());

  input! {r: usize, x: usize, y: usize}

  let distance = ((x.pow(2) + y.pow(2)) as f64).sqrt();
  let steps = (distance / (r as f64)).ceil();
  println!("{} {} {}", distance, distance % (r as f64), steps);
  let total = if distance % (r as f64) == 0f64 || steps > 1f64 {
    steps
  } else {
    steps + 1.0f64
  };
  writeln!(out, "{}", total).unwrap();
}

File: java/abc198/C.java
import java.util.*;

public class C {
  public static void main(String[] args) {
    Scanner sc = new Scanner(System.in);
    double r = sc.nextDouble();
    double x = sc.nextDouble();
    double y = sc.nextDouble();

    var distance = Math.sqrt(Math.pow(x, 2.0) + Math.pow(y, 2.0));
    var step = Math.ceil(distance / r);

    if (distance % r == 0 || step > 1) {
      System.out.println((int) step);
    } else {
      System.out.println((int) step + 1);
    }
  }
}
